### DISCUSSION TITLE
Issue with Metamask connect

### DISCUSSION THREAD

**Discussion by ngmisl** [jeffreylanters/react-unity-webgl#419] [Help Wanted!] [answered]
Hopefully someone can help me here, as I've really hit a wall:

added unity to react via react-unity-webgl it works, but upon hitting connect to connect with metamask I get this weird error. game works perfectly fine in standalone... anyone got any idea?

![image](https://user-images.githubusercontent.com/98217124/181502122-2d9b8f75-8392-461a-a689-7797a81f1417.png)

### Component

```
import React from "react";
import { Unity, useUnityContext } from "react-unity-webgl";

const Mint = () => {
  const { unityProvider } = useUnityContext({
    loaderUrl: "../build/Test21.loader.js",
    dataUrl: "../build/Test21.data",
    frameworkUrl: "../build/Test21.framework.js",
    codeUrl: "../build/Test21.wasm",
  });
  return (
    <Unity
      unityProvider={unityProvider}
      style={{ width: 1000, height: 1000 }}
    />
  );
};

export default Mint;
```

**Comment by jeffreylanters** [comment, marked as answer]
When using a WebGL plugin in your Unity project, it may contain custom JavaScript. This is usually inserted into the output HTML. 

When using React Unity WebGL, your only using the actual Unity build, and are not including the build's HTML thus losing the JavaScript included with it. This is why you're getting this error.

A solution would be to move the build's external JavaScript into your project, and include it into your React's base HTML.

**Reply by ngmisl** [reply]
thanks a ton! works now :)

**Reply by jeffreylanters** [reply]
Awesome! Happy coding!

**Comment by BestFriend0523** [comment]
I also have same issue.

**Comment by jeffreylanters** [comment]
Hi @DarlingUUi! My guess is that it has to do with the fact that your send message is invoked during an asynchronous callback in the "then" closure. This means that it's possible for the component to have been rerendered in the meanwhile, without the reference to the SendMessage callback to be updated since it's out of scope.

This would be a problem in any React project. A solution would be to move the actual trigger of the method to an Effect hook.
